### PR TITLE
test: add integration tests to bestool

### PR DESCRIPTION
**Pull request by rail-rain**
Right now, it's just a rough demo of `trycmd` with one test case for Tamanu Alerts.

### CODE REVIEW

**Comment by passcod** [comment]
love it! would be great to disable logging when testing as that's unlikely to ever be useful test output

**Comment by rail-rain** [comment]
Would you disable all logging? I thought warnings may be useful to test. For example, when a `shell` alert source times out, a warning-level log is the only way to know that it timed out. I agree that testing info-level logs is mostly useless.

**Comment by passcod** [comment]
good call. limiting to warning and up by default at least would be best; also would be nice to find a way to kill colour usage for tests so it doesn't have all the [33m etm escapes

File: crates/bestool/tests/cli_tests.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+fn cli_tests() {
+	let cases = trycmd::TestCases::new();
+	cases
+		.env("BESTOOL_MOCK_TIME", "1")
+		.env("NO_COLOR", "1")
+		.case("tests/cmd/*.toml");
+
+	let handle_res = init_db().and_then(run_db);
+
+	if matches!(handle_res, Err(_)) {
```
Comment (passcod) [inline crates/bestool/tests/cli_tests.rs:16]:
```suggestion
	if handle_res.is_err() {
```

**Review by passcod — APPROVED** [review]

File: crates/bestool/tests/cli_tests.rs [hunk 2]
```diff
@@ -110,3 +117,73 @@ fn stop_db(temp_dir: TempDir) -> Result<()> {
 
 	Ok(())
 }
+
+pub fn find_postgres_bin(name: &str) -> Result<OsString> {
```
Comment (passcod) [inline crates/bestool/tests/cli_tests.rs:121]: is this copied from the lib? can we expose that and re-use instead?

Comment (rail-rain) [reply crates/bestool/tests/cli_tests.rs]: Oh, I missed I can link the `test` crate to `bestool` because trycmd doesn't use that. It's fixed now.

**Comment by rail-rain** [comment]
Tests that require Postgres now works on Ubuntu and Windows runners (which have Postgres installed by default). On the Mac runner, those get ignored.

**Comment by passcod** [comment]
Nice!

**Pull request merged**